src/docs/backend/cross-region-rpc.mdx

```diff
@@ -0,0 +1,157 @@
+
+- `ByOrganizationSlug` will extract the `organization_slug` parameter and use it to locate the region using `sentry_organizationmapping`.
+- `ByOrganizationId` will extract the `organization_id` parameter and use it to locate the organization’s region using `sentry_organizationmapping`
+- `ByRegionName` uses the `region_name` parameter to choose a destination region.
+- `ByOrganizationIsAttribute(param)` Will extract `organization_id` from `param` to resolve the region.
+
+The implementation for `get_org_by_slug` looks like:
+
+```python
+from sentry.services.hybrid_cloud.organization.service import OrganizationService
+from sentry.services.hybrid_cloud.organization.model import RpcOrganizationSummary
+from sentry.models.organization import Organization, OrganizationStatus
+
+class DatabaseBackedOrganizationService(OrganizationService):
+
+    def get_org_by_slug(
+        self,
+        *,
+        slug: str,
+        user_id: int | None = None,
+    ) -> RpcOrganizationSummary | None:
+        query = Organization.objects.filter(slug=slug)
+        if user_id is not None:
+            query = query.filter(
+                status=OrganizationStatus.ACTIVE,
+                member_set__user_id=user_id,
+            )
+        try:
+            return serialize_organization_summary(query.get())
+        except Organization.DoesNotExist:
+            return None
+```
+
+RPC method implementations are simple python methods, but there are a few rules that should be followed:
+
+1. All parameters and return values must either be simple values, or `RpcModel` instances. This ensures that requests and responses can be serialized.
+2. Response values should be scalar values or `RpcModel` instances. Avoid tuples or lists that are not sequences of objects.
+
+### RPC Authentication
+
+All cross-region RPC requests go to a single endpoint `/api/0/internal/rpc/:service/:method` The RPC endpoint uses HMAC based signature comparison to validate authenticity. Before a request is made, the request body has an HMAC signature created using a secret that is shared by region and control silo instances.
+
+When an RPC message is received, the signature header is compared with a locally generated HMAC using the receiver’s secret. The request is only processed if the checksums match.
+
+### RPC serialization and transport
+
+Cross-Region RPC uses pydantic and openapi to provide typehinted parameter and response values, and fulfil our requirements for schema generation and validation. How RPC methods behave varies based on how the application is configured.
+
+#### In monolith mode
+
+- RPC methods are invoked synchronously on the implementation service.
+
+#### In tests
+
+- During tests we emulate as much of the RPC stack as we can.
+- Request bodies and responses are serialized as JSON.
+- Requests are made with `django.test.Client` to the RPC endpoint.
+
+#### In siloed mode
```

Comment:
```suggestion
#### In siloed mode
```
Not sure if it's best to mention in the monolith mode section or here but might be worth mentioning the siloed mode case where the destination is local and no RPC request is actually made

Reply:
I think it works here - added.